.github/actions/constellation_recover/action.yml

```diff
@@ -0,0 +1,102 @@
+name: Constellation recover
+description: "Recover a Constellation cluster with an unavailable control plane."
+
+inputs:
+  kubeconfig:
+    description: "The kubeconfig for the cluster."
+    required: true
+  masterSecret:
+    description: "The master-secret for the cluster."
+    required: true
+  cloudProvider:
+    description: "Which cloud provider to use."
+    required: true
+  gcpProject:
+    description: "The GCP project Constellation is deployed in."
+    required: false
+  resourceGroup:
+    description: "The Azure resource group Constellation is deployed in."
+    required: false
+
+runs:
+  using: "composite"
+  steps:
+    # Get Constellation UID
+    - name: Get Constellation cluster UID
+      shell: bash
+      run: |
+        CONSTELLATION_UID=$(yq -r ".uid" constellation-id.json)
+        echo "CONSTELLATION_UID=$CONSTELLATION_UID" >> $GITHUB_ENV
+      env:
+        KUBECONFIG: ${{ inputs.kubeconfig }}
+    # Restart worker node machine and wait for it to rejoin the cluster
+    - name: Restart worker node on GCP
+      if: ${{ inputs.cloudProvider == 'gcp' }}
+      shell: bash
+      run: |
+        export GCP_ZONE=$(yq -r ".provider.gcp.zone" constellation-conf.yaml)
+        export WORKER_NODE_INSTANCE=$(gcloud compute instances list --project=${{ inputs.gcpProject }} --zones=${GCP_ZONE} --filter="labels.constellation-role=worker AND labels.constellation-uid=${CONSTELLATION_UID}" --format="json" | jq '.[0].name' -r)
+        echo Shutting down worker instance: ${WORKER_NODE_INSTANCE}
+        gcloud compute instances reset ${WORKER_NODE_INSTANCE} --project=${{ inputs.gcpProject }} --zone=${GCP_ZONE}
+        kubectl wait --for=condition=ready --all nodes --timeout=10m
+      env:
+        KUBECONFIG: ${{ inputs.kubeconfig }}
+    - name: Restart worker node on Azure
+      if: ${{ inputs.cloudProvider == 'azure' }}
+      shell: bash
+      run: |
+        echo Shutting down worker instance: e2e-test-${CONSTELLATION_UID}-worker
+        az vmss restart --instance-ids 0 --name e2e-test-${CONSTELLATION_UID}-worker --resource-group ${{ inputs.resourceGroup }}
+        kubectl wait --for=condition=ready --all nodes --timeout=10m
```

Comment:
I am not sure, but is the K8s API quick enough to see the node as restarting (i.e. unhealthy) for this to actually wait for the restart to finish? I could image this check going right through without ever seeing the node go unhealthy. 

I think a more robust assert should wait for a node to show up as unhealthy, and then healthy again.

Reply:
It did so on my local tests, but I get the point. I think about doing something like:
```
kubectl wait --for=condition=Ready=false --timeout=10m e2e-test-${CONSTELLATION_UID}-worker
kubectl wait --for=condition=Ready=true --timeout=10m --all nodes```